spec_synapse_v2.md

```diff
@@ -369,6 +369,40 @@ datatype: FLOAT
 unit: micro-meters
 
 
+### morpho_branch_order_pre
+description: Branch order of the section associated with the pre-synaptic neuron.
+Used in detailed model.
+
+This property is optional
+
+dataset size: 1xS
+
+datatype: INT64
+
+
+### morpho_branch_order_post
+description: Branch order of the section associated with the post-synaptic neuron (-1 for on-soma synapses).
+Used in detailed model.
+
+This property is optional
+
+dataset size: 1xS
+
+datatype: INT64
+
+
+### morpho_branch_type_post
+description: Branch type of the section associated with the post-synaptic neuron.
+0 stands for soma, 1 for axon, 2 for basal dendrite and 3 for apical dendrite.
```

Comment:
Yes, you are right that these are redundant (as center positions are, in fact).

However they are:
a) used as one of the synapse properties for different types of analysis
b) costly to retrieve every time since it requires loading morphologies

That's why they ended up being in NRN, I suppose.
Since SYN2 is designed in a way that adding a dataset should not affect the performance; and storing them should be relatively cheap (int ones, at least), I think it's as worth having them for detailed morphologies as it is worth storing precomputed positions.